contracts/test/v0.8/functions/v0/Gas.test.ts

```diff
@@ -0,0 +1,158 @@
+import { ethers } from 'hardhat'
+import { expect } from 'chai'
+import { Contract, ContractFactory } from 'ethers'
+import { Roles, getUsers } from '../../../test-helpers/setup'
+import { stringToBytes } from '../../../test-helpers/helpers'
+
+let concreteFunctionsClientFactory: ContractFactory
+let functionsOracleFactory: ContractFactory
+let functionsBillingRegistryFactory: ContractFactory
+let linkTokenFactory: ContractFactory
+let mockAggregatorV3Factory: ContractFactory
+let roles: Roles
+
+function getEventArg(events: any, eventName: string, argIndex: number) {
+  if (Array.isArray(events)) {
+    const event = events.find((e: any) => e.event == eventName)
+    if (event && Array.isArray(event.args) && event.args.length > 0) {
+      return event.args[argIndex]
+    }
+  }
+  return undefined
+}
+
+const baselineGasUsed = 641560
+let currentGasUsed = 0
+
+before(async () => {
+  roles = (await getUsers()).roles
+
+  concreteFunctionsClientFactory = await ethers.getContractFactory(
+    'src/v0.8/functions/tests/0_0_0/testhelpers/FunctionsClientTestHelper.sol:FunctionsClientTestHelper',
+    roles.defaultAccount,
+  )
+  functionsOracleFactory = await ethers.getContractFactory(
+    'src/v0.8/functions/tests/0_0_0/testhelpers/FunctionsOracleHelper.sol:FunctionsOracleHelper',
+    roles.defaultAccount,
+  )
+
+  functionsBillingRegistryFactory = await ethers.getContractFactory(
+    'src/v0.8/functions/tests/0_0_0/testhelpers/FunctionsBillingRegistryWithInit.sol:FunctionsBillingRegistryWithInit',
+    roles.defaultAccount,
+  )
+
+  linkTokenFactory = await ethers.getContractFactory(
+    'src/v0.4/LinkToken.sol:LinkToken',
+    roles.consumer,
+  )
+
+  mockAggregatorV3Factory = await ethers.getContractFactory(
+    'src/v0.7/tests/MockV3Aggregator.sol:MockV3Aggregator',
```

Comment:
there are newer MockV3Aggregator, generally never import outside of the v0.8 folder